File: src/main/java/mc/rpgstats/component/internal/PlayerHealthAttachComponent.java
package mc.rpgstats.component.internal;

import dev.onyxstudios.cca.api.v3.component.Component;
import mc.rpgstats.main.RPGStats;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.nbt.NbtCompound;
import net.minecraft.server.network.ServerPlayerEntity;
import org.jetbrains.annotations.NotNull;

public class PlayerHealthAttachComponent implements Component {
    public PlayerEntity playerEntity;
    
    public PlayerHealthAttachComponent(PlayerEntity playerEntity) {
        this.playerEntity = playerEntity;
    }
    
    @Override
    public void readFromNbt(@NotNull NbtCompound compoundTag) {
        if (playerEntity instanceof ServerPlayerEntity) {
            RPGStats.needsStatFix.add((ServerPlayerEntity)playerEntity);
        }
    }
    
    @Override
    public void writeToNbt(@NotNull NbtCompound compoundTag) {}
}

File: src/main/kotlin/io/github/silverandro/rpgstats/stats/internal/PlayerHealthAttachComponent.kt
package io.github.silverandro.rpgstats.stats.internal

import dev.onyxstudios.cca.api.v3.component.Component
import io.github.silverandro.rpgstats.Events
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.nbt.NbtCompound
import net.minecraft.server.network.ServerPlayerEntity

class PlayerHealthAttachComponent(var playerEntity: PlayerEntity) : Component {
    override fun readFromNbt(compoundTag: NbtCompound) {
        if (playerEntity is ServerPlayerEntity) {
            Events.needsStatFix.add(playerEntity as ServerPlayerEntity)
        }
    }

    override fun writeToNbt(compoundTag: NbtCompound) {}
}